include/dali/imgcodec/image_format.h

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_IMGCODEC_IMAGE_FORMAT_H_
+#define DALI_IMGCODEC_IMAGE_FORMAT_H_
+
+#include <map>
+#include <memory>
+#include <stdexcept>
+#include <string>
+#include <vector>
+#include "dali/core/span.h"
+#include "dali/core/tensor_shape.h"
+#include "dali/core/stream.h"
+#include "dali/imgcodec/image_source.h"
+
+namespace dali {
+namespace imgcodec {
+
+struct ImageInfo {
+  TensorShape<> shape;
+  struct {
+    int rotate;
+    bool flip_x;
+    bool flip_y;
+  } orientation;
+};
+
+class ImageParser {
+ public:
+  virtual ~ImageParser() = default;
+  virtual ImageInfo GetInfo(ImageSource *encoded) const = 0;
+  virtual bool CanParse(ImageSource *encoded) const = 0;
+};
+
+class ImageCodec;
+
+class DLL_PUBLIC ImageFormat {
+ public:
+  ImageFormat(const char *name, shared_ptr<ImageParser> parser);
+
+  bool Matches(ImageSource *encoded) const;
+  ImageParser* Parser() const;
+  const std::string& Name() const;
+  span<ImageCodec* const> Codecs() const;
+  void RegisterCodec(std::shared_ptr<ImageCodec> codec, float priority);
+
+ private:
+  std::string name_;
+  std::shared_ptr<ImageParser> parser_;
+  std::multimap<float, std::shared_ptr<ImageCodec>> codecs_;
+  std::vector<ImageCodec*> codec_ptrs_;
+};
+
+class DLL_PUBLIC ImageFormatRegistry {
```

Comment:
It's not exactly _the_ entry point - there must be something on a higher level, but we're still considering whether that something should be a part of this library or left to the operator.

Reply:
I meant entry point to the part that is in this PR. 

I am asking about this because I am a bit afraid that this design is incomplete with regards on how it will be integrated with DALI. This will probably cause some, maybe significant changes, when this integration eventually comes. 

I think it would be good to have that figured out before we spend some significant effort into implementing it for particular formats.